packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -172,25 +192,101 @@ class BottomNavigationBar extends StatefulWidget {
   /// The index into [items] of the current active item.
   final int currentIndex;
 
+  /// The z-coordinate of this [BottomNavigationBar].
+  ///
+  /// If null, defaults to `8.0`.
+  ///
+  /// {@macro flutter.material.material.elevation}
+  final double elevation;
+
   /// Defines the layout and behavior of a [BottomNavigationBar].
   ///
   /// See documentation for [BottomNavigationBarType] for information on the meaning
   /// of different types.
   final BottomNavigationBarType type;
 
+  /// Deprecated, use [selectedItemColor].
+  ///
   /// The color of the selected item when bottom navigation bar is
   /// [BottomNavigationBarType.fixed].
   ///
   /// If [fixedColor] is null then the theme's primary color,
   /// [ThemeData.primaryColor], is used. However if [BottomNavigationBar.type] is
   /// [BottomNavigationBarType.shifting] then [fixedColor] is ignored.
+  ///
+  /// If [selectedItemColor] is provided, it will be used over this.
   final Color fixedColor;
 
+  /// The color of the [BottomNavigationBar] itself.
+  ///
+  /// If [type] is [BottomNavigationBarType.shifting] and the
+  /// [items]s, have [BottomNavigationBarItem.backgroundColor] set, the [item]'s
+  /// backgroundColor will splash and overwrite this color.
+  final Color backgroundColor;
+
   /// The size of all of the [BottomNavigationBarItem] icons.
   ///
   /// See [BottomNavigationBarItem.icon] for more information.
   final double iconSize;
 
+  /// The color of the selected [BottomNavigationBarItem.icon] + [BottomNavigationBarItem.label].
+  ///
+  /// If null then the [ThemeData.primaryColor] is used.
+  ///
+  /// If provided, this will take precedence over [fixedColor].
+  final Color selectedItemColor;
+
+  /// The color of the unselected [BottomNavigationBarItem.icon] + [BottomNavigationBarItem.label]s.
+  ///
+  /// If null then the [TextTheme.caption]'s color is used.
+  final Color unselectedItemColor;
+
+  /// The font size of the [BottomNavigationBarItem] labels when they are selected.
+  ///
+  /// Defaults to `14.0`.
+  final double selectedFontSize;
+
+  /// The font size of the [BottomNavigationBarItem] label when they are not selected.
+  ///
+  /// Defaults to `12.0`.
+  final double unselectedFontSize;
+
+  /// Whether the labels are shown for the selected [BottomNavigationBarItem].
+  final bool showUnselectedLabels;
+
+  /// Whether the labels are shown for the unselected [BottomNavigationBarItem]s.
+  final bool showSelectedLabels;
+
+  /// Used by the [BottomNavigationBar] constructor to set the [type] parameter.
+  ///
+  /// If type is provided, it is returned. Otherwise,
+  /// [BottomNavigationBarType.fixed] is used for 3 or fewer items, and
+  /// [BottomNavigationBarType.shifting] is used for 4+ items.
+  static BottomNavigationBarType _type(
```

Comment:
Since this is private these should be`//` comments.

Reply:
Done.